src/plugins/topic_viewer/TopicViewer.cc

```diff
@@ -0,0 +1,301 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+#include "TopicViewer.hh"
+#include <ignition/plugin/Register.hh>
+
+namespace ignition
+{
+namespace gui
+{
+namespace plugins
+{
+  class TopicsModel : public QStandardItemModel
+  {
+    /// \brief roles and names of the model
+    public: QHash<int, QByteArray> roleNames() const override
+    {
+      QHash<int, QByteArray> roles;
+      roles[NAME_ROLE] = NAME_KEY;
+      roles[TYPE_ROLE] = TYPE_KEY;
+      roles[TOPIC_ROLE] = TOPIC_KEY;
+      roles[PATH_ROLE] = PATH_KEY;
+      roles[PLOT_ROLE] = PLOT_KEY;
+      return roles;
+    }
+  };
+
+  class TopicViewerPrivate
+  {
+    /// \brief Node for Commincation
+    public: ignition::transport::Node node;
+
+    /// \brief Model to create it from the available topics and messages
+    public: TopicsModel *model;
+  };
+}
+}
+}
+using namespace ignition;
+using namespace gui;
+using namespace plugins;
+
+TopicViewer::TopicViewer() :Plugin(), dataPtr(new TopicViewerPrivate)
+{
+  using namespace google::protobuf;
+  this->plotableTypes.push_back(FieldDescriptor::Type::TYPE_DOUBLE);
+  this->plotableTypes.push_back(FieldDescriptor::Type::TYPE_FLOAT);
+  this->plotableTypes.push_back(FieldDescriptor::Type::TYPE_INT32);
+  this->plotableTypes.push_back(FieldDescriptor::Type::TYPE_INT64);
+  this->plotableTypes.push_back(FieldDescriptor::Type::TYPE_UINT32);
+  this->plotableTypes.push_back(FieldDescriptor::Type::TYPE_UINT64);
+  this->plotableTypes.push_back(FieldDescriptor::Type::TYPE_BOOL);
+
+  this->plottingMode = true;
+
+  this->CreateModel();
+
+  ignition::gui::App()->Engine()->rootContext()->setContextProperty(
+                "TopicsModel", this->dataPtr->model);
+}
+
+TopicViewer::~TopicViewer()
+{
+}
+
+void TopicViewer::LoadConfig(const tinyxml2::XMLElement *)
+{
+  if (this->title.empty())
+    this->title = "Topic Viewer";
+}
+
+void TopicViewer::CreateModel()
+{
+  this->dataPtr->model = new TopicsModel();
+
+  std::vector<std::string> topics;
+  this->dataPtr->node.TopicList(topics);
+
+  for (unsigned int i = 0; i < topics.size(); i++)
+  {
+    std::vector<ignition::transport::MessagePublisher> infoMsgs;
+    this->dataPtr->node.TopicInfo(topics[i], infoMsgs);
+    std::string msgType = infoMsgs[0].MsgTypeName();
+    this->AddTopic(topics[i], msgType);
+  }
+}
+
+//////////////////////////////////////////////////
+void TopicViewer::SetPlottingMode(bool _mode)
+{
+  this->plottingMode = _mode;
+}
+
+bool TopicViewer :: GetPlottingMode()
+{
+  return this->plottingMode;
+}
+
+//////////////////////////////////////////////////
+void TopicViewer::AddTopic(std::string _topic, std::string _msg)
+{
+  QStandardItem *topicItem = this->FactoryItem(_topic, "Topic");
+  QStandardItem *parent = this->dataPtr->model->invisibleRootItem();
+  parent->appendRow(topicItem);
+
+  // remove 'ignition.msgs.' from msg name
+  try {
```

Comment:
```suggestion
  try
  {
```
style guide #3: braces need to be on their own lines